Description: add support for loongarch64
 .
 cfitsio (4.3.0-1+loong64) unreleased; urgency=medium
 .
   * Add support for loongarch64.
Author: Dandan Zhang <zhangdandan@loongson.cn>

---
Last-Update: 2023-09-07

--- cfitsio-4.3.0.orig/fitsio2.h
+++ cfitsio-4.3.0/fitsio2.h
@@ -120,6 +120,17 @@ extern int Fitsio_Pthread_Status;
 #  endif
 #  define LONGSIZE 64
 
+#elif defined(__loongarch__)
+#define BYTESWAPPED TRUE
+#  if __loongarch_grlen == 32
+#    define LONGSIZE 32
+#  elif __loongarch_grlen == 64
+#    define LONGSIZE 64
+#  else
+#    error "can't handle long size given by __loongarch_grlen"
+#  endif
+
+
 #elif defined(_MIPS_SZLONG)
 
 #  if defined(MIPSEL)
